<a href="https://colab.research.google.com/github/tanpaterusan/data-science-2026/blob/main/Pertemuan2_Amalia_240401010284.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np

In [2]:
# 1 Muat Dataset Titanic
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)

In [3]:
# 2 Eksplorasi Awal
print(df.shape, df.columns.tolist())
print(df.isnull().sum())

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


Kuis 1:
Berapa jumlah baris dan kolom dataset Titanic?

> 891 baris dan 12 kolom.




Kuis 2: Kolom mana yang memiliki nilai kosong (missing)?


> Age sebanyak 177 baris, Cabin sebanyak 687 baris dan Embarked sebanyak 2 baris.



In [4]:

# 3 Filtering — Penumpang Wanita Kelas 1
wk1 = df[(df['Sex']=='female') &
         (df['Pclass']==1)]
print(wk1['Survived'].mean(), wk1['Age'].mean())

0.9680851063829787 34.61176470588235


Kuis 3: Rata-rata usia penumpang wanita kelas 1 yang selamat?


> 34 tahun.





In [5]:
# 4 Groupby — Tingkat Keselamatan per Kelas
hasil = df.groupby('Pclass').agg(
    total=('PassengerId','count'),
    selamat=('Survived','mean'),
).round(3)
print(hasil)

        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


Kuis 4: Kelas penumpang mana yang tingkat keselamatan tertinggi?


> Penumpang kelas 3



In [6]:
# Kuis 5: Tampilkan hasil groupby('Pclass')['Survived'].mean()

print(df.groupby('Pclass')['Survived'].mean())

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64
